### ISSUE TITLE
Bug in Implementation of PGAN

### ISSUE THREAD

**Issue by StefanFischer** [azadis/SB-GAN#9]
Hey,

i found a bug in the implementation of your progressive GAN.
In both implementations of your gadient_penalty (progressive.py and gan_loss.py)
the alpha value for smooth integration of the next resolution layer is overwritten for the alpha of the gradient penalty.

``````
def gradient_penalty(self, x_real, x_fake, interpolate, alpha):
    n = x_real.size(0)
    device = x_real.device
    alpha = torch.rand(n)
    alpha = alpha.to(device)
    alpha = alpha[:, None, None, None]
    interpolates = alpha * x_real.detach() + (1 - alpha) * x_fake.detach()
``````

This can affect the training with shocks in lower resolutions if this alpha is set high as states by Terras et al. in their paper to PGAN.
